HealthMonitoring/Model/EndpointMetadata.cs

```diff
@@ -1,18 +1,33 @@
+using System;
 using System.Linq;
 
 namespace HealthMonitoring.Model
 {
     public class EndpointMetadata
     {
-        public EndpointMetadata(string name, string group, string[] tags)
+        public EndpointMetadata(string name, string group, string[] tags, DateTime? firstTimeRegistered = null, DateTime? lastTimeRegistrationUpdated = null)
```

Comment:
it could be null in few cases, so first is when we registered endpoint only once, last registration update should be null, other way would be misleading  (if we update both) i guess. And second for existing endpoints we dont know when they were registered initially, but this one is probably not such important as it makes no sense to keep it as a null and we can provide default value for that. 
But anyway we can make those fields not nullable